lithops/executors.py

```diff
@@ -511,6 +515,41 @@ def get_result(self, fs=None, throw_except=True, timeout=None,
 
         return result
 
+    def job_summary(self, cloud_objects_n=0):
+        """
+        logs information of a job executed by the calling function executor.
+        currently supports: code_engine, ibm_vpc and ibm_cf.
+
+        :param cloud_objects_n: number of cloud object used by user in COS.
+        """
+
+        def append(content):
+            """ appends information processed to a log file."""
+            headers = ['Function Name', 'Invocations', 'Memory Used(MB)',
+                       'Average Runtime(sec)', 'Cost($)', 'Cloud Objects Used']
+            pd.DataFrame(content).to_csv(path, mode='a', header=headers, index=False)
+
+        def calc_cost(total_runtime, memory_gb):
+            """ returns total cost associated with executing the calling function-executor's job."""
+            return self.config['unit_price'] * memory_gb * total_runtime
+
+        # avoid logging info unless pricing comes attached to chosen computational backend.
+        if 'unit_price' in self.config:
+
+            path = os.path.join(constants.LOGS_DIR, datetime.now().strftime("%Y-%m-%d_%H:%M:%S.csv"))
+            futures = self.futures
+            if type(futures) != list:
+                futures = [futures]
+            actions_num = len(futures)
+            func_name = futures[0].function_name
+            memory_mb = futures[0].runtime_memory  # memory required to run a single invocation
```

Comment:
In Lithops you can run this:
```python
fexec = lithops.FunctionExecutor()
fexec.map(my_map_function, range(1), runtime_memory=4096)
fexec.map(my_map_function, range(1000), runtime_memory=128)
```
If you take into account only the first future, the calculated amount could be wrong.
I think you should group `runtimes` by `runtime_memory`, maybe into a dict, and then pass this dict to the `calc_cost(runtimes_dict)` function.

Reply:
Appreciate the detailed input Josep!